### PR TITLE
Allocate output in case of rasterisation of partially-decoded DC

### LINKED ISSUE #254: Tests fail on ASAN build
**Describe the bug**
```
        1576 - DecodeTest.PixelTestOpaqueSrgbLossyNoise (Failed)
        1637 - DecodeTest/DecodeTestParam.PixelTest/301x33RGBtoRGBu8#GetParam()=301x33RGBtoRGBu8 (Failed)
        1661 - DecodeTest/DecodeTestParam.PixelTest/301x33RGBtoRGBAu8#GetParam()=301x33RGBtoRGBAu8 (Failed)
        1957 - RoundtripTest.Uint8FrameRoundtripTest (Failed)
        1960 - RoundtripTest.TestICCProfile (Failed)
```

**To Reproduce**
/ci.sh asan

### PR DESCRIPTION
**Pull request by eustas**

### CODE REVIEW

**Review by veluca93 — APPROVED** [review]

File: lib/jxl/dec_frame.cc [hunk 1]
```diff
@@ -909,6 +909,12 @@ Status FrameDecoder::FinalizeFrame() {
         "FinalizeFrame called before the frame was fully decoded");
   }
 
+  if (!finalized_dc_) {
+    JXL_DASSERT(allow_partial_frames_);
```
Comment (veluca93) [inline lib/jxl/dec_frame.cc:913]: Are we sure that this DASSERT cannot be triggered by erraneous usage?

Comment (jonsneyers) [reply lib/jxl/dec_frame.cc:913]: Probably best to use regular JXL_ASSERT in cases like this where it's not in a hot loop — makes it more likely to catch things.

Comment (eustas) [reply lib/jxl/dec_frame.cc:913]: Yup, will change to DASSERT.
decoded_ac_global_ checked in preceding `if` is set to true at the end of ProcessACGlobal that is invoked if finalized_dc_. This assertion is more for readability than for safety.

Comment (eustas) [reply lib/jxl/dec_frame.cc]: PTAL

**Review by veluca93 — APPROVED** [review]

**Pull request merged**